packages/flutter/lib/src/material/text_field.dart

```diff
@@ -79,6 +83,13 @@ class _TextFieldSelectionGestureDetectorBuilder extends TextSelectionGestureDete
           );
           break;
       }
+      if (_state._effectiveController.selection != _lastSelection) {
```

Comment:
The `HapticFeedback` should only trigger if  `onSingleLongTapMoveUpdate` changes the selection I think? So I don't think you need the 2 extra instance variables you can just get the selection at the start of this method and check if it has changed?

Reply:
I thought this didn't work when I tried it but turns out it actually does